api/control/v1alpha1/types.go

```diff
@@ -665,3 +665,56 @@ type CustomTransformList struct {
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []CustomTransform `json:"items"`
 }
+
+// ContentClassifier overrides the default categorization of the contents of
+// a particular resource.version.apiGroup of workload object.
+// The name of the ContentClassifier must be exactly resource.version.apiGroup
+// (resource.version when the apiGroup is the empty string).
+// This is modeled on what `kubectl` accepts.
+// Examples: pods.v1, jobs.v1.batch, networkpolicies.v1.networking.k8s.io.
+//
+// The contents of each workload object are divided into three categories:
+// desired state (which propagates from core to WEC),
+// reported state (which propagates from WEC to core),
+// and state that does not propagate (e.g., state that is cluster-specific
+// and managed by local servers/controllers).
+//
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,shortName={cc},categories={all}
+type ContentClassifier struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ContentClassifierSpec   `json:"spec"`
+	Status ContentClassifierStatus `json:"status,omitempty"`
+}
+
+// ContentClassifierSpec specifies the overrides.
+// Currently there are limitations:
+// 1. You can not add to the reported state;
+// 2. You can not declare any part of `.status` to be desired state.
+// Invalid overrides will be reported in `.status.errors` and otherwise ignored.
+// Currently the overrides are expressed by identifying object parts
+// by JSON Pointers (RFC 6901).
+type ContentClassifierSpec struct {
+	DesiredStatePointers  []ParsedJSONPointer `json:"desiredStatePointers,omitempty"`
+	DontPropagatePointers []ParsedJSONPointer `json:"dontPropagatePointers,omitempty"`
```

Comment:
Two points:
1. Can both fields be set? If I understood what they specify, then the answer is no. This should be clarified in the struct comment/docs.
2. The name `dontPropagatePointers` is more instructional than declarative in nature. Perhaps it can be changed to something declarative like `restrictedPaths` (or both fields can be declarative counterparts).

Reply:
1. These two fields hold _overrides_ to the default classification; an object field not covered by any override is classified in the default way. Yes, one kind of object can have both categories of overrides.
2. While we are on the subject of naming, I should point out that "desired state" and "reported state" do not get to the relevant point, which is about propagation.